### ISSUE TITLE
[HttpFoundation] Failed to set session handler, headers already sent by phpunit

### ISSUE THREAD

**Issue by anlutro** [symfony/symfony#24524]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | yes
| RFC?             | no
| Symfony version  | 3.4.x

I have a project that depends on symfony/http-foundation. With 3.4.x-dev I'm getting some RuntimeErrors recently:

```
RuntimeException: Failed to set the session handler because headers have already been sent
by ".../vendor/phpunit/phpunit/src/Util/Printer.php" at line 110.

.../vendor/symfony/http-foundation/Session/Storage/NativeSessionStorage.php:397
.../vendor/symfony/http-foundation/Session/Storage/NativeSessionStorage.php:112
.../classes/Http/SessionProvider.php:128
.../vendor/autarky/container/classes/Factory/Factory.php:89
.../vendor/autarky/container/classes/Container.php:256
.../vendor/autarky/container/classes/Container.php:226
.../classes/Application.php:511
.../tests/unit/Session/SessionProviderTest.php:20
```

I tracked down the line to https://github.com/symfony/symfony/pull/23711 which was merged 2 days ago. If I remove the if headers_sent thing from setSaveHandler, my test runs fine again.

PHPUnit 5.7.22, symfony/http-foundation is at 0fe9b3dc9902a89e646ca5bf053690018830999b

**Comment by nicolas-grekas** [comment]
You're going to have this error anyway when running on PHP 7.2, so you should fix your tests.
Running them in a separate process should do the trick.

**Comment by sroze** [comment]
@nicolas-grekas I think the issue is a bit deeper. What's new in PHP 7.2 is that PHP won't allow you **anymore** to change the session's ini settings or save handlers after the session has been started.

This should be fixed in https://github.com/symfony/symfony/pull/24516 (i.e. you will get an exception only if you try to change things while creating the `NativeSessionStorage` object, not every time you create an instance of it 🤞) - but it's blocked by a remaining failing test.

**Comment by garak** [comment]
This should be merged in `3.4` branch too. I'm on a project with php 7.2/Symfony 3.4-beta3 and there's no way to get this working :disappointed:

**Comment by garak** [comment]
Also, I tried a dirty hack, by copying `NativeSessionStorage` class from https://github.com/symfony/symfony/pull/24531, and I'm still getting the same error ("Failed to set...")

**Comment by umpirsky** [comment]
Upgraded to `v3.4.0-BETA3`, same error I reported in https://github.com/symfony/symfony/issues/24800, which is probably a duplicate of this issue.

@nicolas-grekas I think we should reopen.

**Comment by tacman** [comment]
Any sort of hack we can use in the meantime?

**Comment by karser** [comment]
I'm getting this error when `APP_ENV=dev` but the error goes away when `APP_ENV=test`.
Symfony v3.4.0-BETA4

**Comment by nicolas-grekas** [comment]
would anyone be able to provide a short reproducer we could work on?

**Comment by anlutro** [comment]
I did a `composer update` and the errors seem to have disappeared. I can do a `git bisect` to figure out what fixed it, if you want.

**Comment by anlutro** [comment]
Looks like it was fixed in https://github.com/symfony/symfony/pull/24952. Closing.

**Comment by anlutro** [comment]
Maybe not.

```php
<?php
class SessionTest extends PHPUnit_Framework_TestCase {
	public function test_session() {
		$h = new \Symfony\Component\HttpFoundation\Session\Storage\Handler\NativeFileSessionHandler();
	}
}
```
```
1) SessionTest::test_session
ini_set(): Headers already sent. You cannot change the session module's ini settings at this time

/usr/src/myapp/vendor/symfony/http-foundation/Session/Storage/Handler/NativeFileSessionHandler.php:52
/usr/src/myapp/tests/SessionTest.php:5
```

Should I re-open or create a new issue? It seems like almost the same issue.
